velox/type/LongDecimal.h

```diff
@@ -57,9 +59,30 @@ struct LongDecimal {
     return unscaledValue_ <= other.unscaledValue_;
   }
 
+  LongDecimal& operator+=(const LongDecimal& other) {
+    unscaledValue_ += other.unscaledValue_;
+    return *this;
+  }
+
+  LongDecimal& operator+=(const ShortDecimal& other) {
+    unscaledValue_ += other.unscaledValue();
+    return *this;
+  }
+
  private:
   int128_t unscaledValue_;
 }; // struct LongDecimal
+
+static inline LongDecimal operator*(const int value, const LongDecimal& other) {
+  return LongDecimal(value * other.unscaledValue());
+}
+
+inline ShortDecimal::ShortDecimal(LongDecimal& other) {
```

Comment:
It is strange to define ShortDecimal constructor in a LongDecimal header file.

Reply:
I wanted this in a header so that it can be inlined. We forward declare `LongDecimal` inside `ShortDecimal.h`. I felt we can only define it here.

Reply:
I wonder if we can avoid defining this constructor altogether. Would you help me understand better why is this needed?

Reply:
I explained the reason below here https://github.com/facebookincubator/velox/pull/2163/files#r935545213